aten/src/ATen/MemoryOverlap.cpp

```diff
@@ -63,7 +63,7 @@ MemOverlapStatus get_overlap_status(const TensorImpl* a, const TensorImpl* b) {
   if (a->numel() == 0 || b->numel() == 0) {
     return MemOverlapStatus::No;
   }
-  if (!a->is_non_overlapping_and_dense() || !b->is_non_overlapping_and_dense()) {
+  if (!a->is_non_overlapping_and_dense_or_false() || !b->is_non_overlapping_and_dense_or_false()) {
```

Comment:
This looks suspicious because the negation. Suppose that a triggers a DDE. Then its conditional will be !false => true, and then you fail to go into the TooHard case!

Reply:
i think thats alright, a true will result into the TooHard path. 
```
  if (!a->is_non_overlapping_and_dense_or_false() || !b->is_non_overlapping_and_dense_or_false()) {
    return MemOverlapStatus::TooHard;
  }
  ```